### PR TITLE
Adding issues for all quarantined tests

### PR DESCRIPTION
**Pull request by BrennanConroy**
* Filed new issues or linked to current issues for all `[QuarantinedTest]` tests.
* Filed issues for tests that pointed to aspnetcore-internal and changed the urls
* Unquarantined a few more tests that I noticed while doing the above
* Unquarantined `InjectedStartup_DefaultApplicationNameIsEntryAssembly` as it's passed for 30+ days and James added some improvements in this area.

### CODE REVIEW

**Review by captainsafia — APPROVED** [review]

**Review by jkotalik — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Sizeable update, removed the quarantine attribute from a few more tests.

**Review by halter73 — APPROVED** [review]

**Comment by halter73** [comment]
Do we have issues for every usage of QuarantinedTestAttribute now? If so, we should remove the default value from the ctor:

https://github.com/dotnet/aspnetcore/blob/540680af5f1d81871d600006d1f6df58492ae409/src/Testing/src/xunit/QuarantinedTestAttribute.cs#L52

**Comment by BrennanConroy** [comment]
> Do we have issues for every usage of QuarantinedTestAttribute now? If so, we should remove the default value from the ctor:

I didn't update the ones that test the attribute works.

**Comment by BrennanConroy** [comment]
> we should remove the default value from the ctor

Done 😄

File: src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs [hunk 1]
```diff
@@ -110,7 +110,7 @@ public static IEnumerable<object[]> LargeUploadData
         }
         [Theory]
         [MemberData(nameof(LargeUploadData))]
-        [QuarantinedTest] // This is inherently flaky and should never be unquarantined.
+        [QuarantinedTest("")] // This is inherently flaky and should never be unquarantined.
```
Comment (Tratcher) [inline src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:113]:
```suggestion
        [QuarantinedTest("This is inherently flaky and should never be unquarantined.")]
```

File: src/Testing/src/xunit/QuarantinedTestAttribute.cs [hunk 2]
```diff
@@ -48,8 +48,8 @@ public sealed class QuarantinedTestAttribute : Attribute, ITraitAttribute
         /// <summary>
         /// Initializes a new instance of the <see cref="QuarantinedTestAttribute"/> class with an optional <see cref="Reason"/>.
         /// </summary>
-        /// <param name="reason">A reason that this test is quarantined.</param>
-        public QuarantinedTestAttribute(string reason = null)
+        /// <param name="reason">A reason that this test is quarantined. Preferably a Github Url.</param>
```
Comment (Tratcher) [inline src/Testing/src/xunit/QuarantinedTestAttribute.cs:51]:
```suggestion
        /// <param name="reason">A reason that this test is quarantined. Preferably a Github issue Url.</param>
```

**Review by Tratcher — APPROVED** [review]

**Pull request merged**